compiler/damlc/daml-doc/src/DA/Daml/Doc/Extract.hs

```diff
@@ -39,13 +47,35 @@ import qualified Data.Text                                 as T
 import           Data.Tuple.Extra                          (second)
 import Data.Either
 
--- | Parse, and process documentation in, a dependency graph of modules.
-mkDocs
-    :: IdeOptions
+-- | Options that affect doc extraction.
+data ExtractOptions = ExtractOptions
+    { eo_qualifyTypes :: QualifyTypes
+        -- ^ qualify non-local types
+    , eo_simplifyQualifiedTypes :: Bool
+        -- ^ drop common module prefix when qualifying types
+    } deriving (Eq, Show, Read)
+
+data QualifyTypes
+    = QualifyTypesAlways
+    | QualifyTypesInPackage
```

Comment:
This option seems weird to me. Why would you want to qualify stuff from the same package but not stuff from other packages?

Reply:
That's why it's an option from many. I think the main reason you'd want it is if your package has some strange convention (...like where you name all your types `T` like in OCaml...), but you still want the standard library (& other packages) to be unqualified like the default.

Reply:
That makes sense. Thanks a lot for the explanation. I have not thought of using the OCaml convention in DAML but I know some people who do use it. :)